### ISSUE TITLE
Accidental multiple selection

### ISSUE THREAD

**Issue by bagp1** [ThePants999/Yellowscribe#20]
I've had an issue I hit a few times before I worked out regarding model selection. When you assign a unit to one data slate, and then without clicking off assign another unit to another data slate, you end up with the first unit still selected and added to the second unit's data slate. Demonstration below.

> https://github.com/ThePants999/Yellowscribe/assets/5026269/df79dee6-2c6e-4d8f-8795-12fbbf10a3eb

The result is the correct dataslate is associated with my wraiths, the correct dataslate is associated with my technomancer, and the wraith model is associated with the technomancer.
> ![image](https://github.com/ThePants999/Yellowscribe/assets/5026269/41dee453-8af6-42d7-ad50-87a3be442187)

The reason I would consider this a bug and not just me using the tool wrong is because the dataslates both worked. If the models can just assign the same way, then the outcome will be correct. 

I haven't looked at the code so I don't yet know what the reasonable fix would be. It could be that model assignment needs to just use the last model selected from a list. Another option would be for group unit assignment to unselect after clicking the yellow machine card, that way you'd have to go out of your way to select the wrong unit for your next assignment.

**Comment by bagp1** [comment]
If either of these fixes are possible given how this app and TTS work respectively, then I'd be able to put in a PR for you. I would appreciate any pointers on getting breakpoints working in VSCode as[ the old instructions](https://docs.google.com/document/d/123M8Qudgy8WzgBOhKIIMqm0dE5djnCgXc3Y77VBF5Tg/edit) reference Atom which was sunsetted and the package servers are down. I also looked at [Moonsharp](https://github.com/tts-community/moonsharp-tts-debug) but I couldn't get it to launch Yellowscribe and I don't really understand how to tell TTS to use the local project files. I can get this issue done without interactive debugging, it would just be nice to have the option.

**Comment by ThePants999** [comment]
Whoops, I somehow missed this issue, sorry!

Yeah, this is an annoying one that I've fallen foul of myself. The trouble is that
- we can't just use the most recent selection because "I want each model to use a random one out of all the models I have selected" is a deliberate feature
- deselecting the currently-selected models after assignment might break people's existing workflows if they don't realise they can multi-select datasheets before selecting models.

Any other spec ideas from anyone welcome!

**Comment by bagp1** [comment]
I generally see what you mean. The random model feature is always quite nice for creating varied armies without hand picking each model.

When you drag select model(s) and then drag select again, you only get the second selection. When you drag select models and then click select a model, you get both selections. If we could make it that the click selection throws away the previous drag selection, that will fix the bug without interfering with the existing workflow you described.

Thanks for maintaining this project; a lot of people (myself included) use and enjoy it.

**Comment by ThePants999** [comment]
Yeah - unfortunately, the selection logic is actual TTS behaviour not mod behaviour :(

**Comment by bagp1** [comment]
Dang, the only other thing I can think of is more visibility over what is selected. Either a preview of the selected models hovering next to the datasheet tile, or even the army being built as you go instead of all at once at the end. The latter sounds like too much work to ever do though.

**Comment by bagp1** [comment]
@ThePants999 I've had a look at the code and think I see the issue. 

At tts-mod\src\Yellow Machine.46ccee.ttslua:345 there is 
```lua
        if #player.getSelectedObjects() == 0 then
            intendedTargets = { player.getHoverObject() }
        else
            intendedTargets = player.getSelectedObjects()

            if not includes(intendedTargets, player.getHoverObject()) then
                table.insert(intendedTargets, player.getHoverObject())
            end
        end
```

Meaning any selected objects are used, but if the player clicked (i.e. hovered) a different model, that gets thrown in too. This is the problem behavior. My opinion is that if the player clicks a model that is not in the previous selection, we should be using the new hover instead of the previous selection e.g.

```lua
        hover_object = player.getHoverObject()
        if hover_object ~= nil then
            intendedTargets = { hover_object }
        else
            intendedTargets = player.getSelectedObjects()
        end
```

This new approach makes more sense because clicking a different model is typically how one deselects a previous selection and selects something new. Throwing in the previous selection was always un-intuitive. The one thing the above probably would break is control click. If we really care about that we could add some handling.

Take the above with a grain of salt because I haven't been able to get YellowMachine to run in my own TTS. Do you have any experience and pointers getting the TTS LUA running?
